burnman/mineral.py

```diff
@@ -64,7 +65,17 @@ class which derives from the equation_of_state base class.
             raise Exception("Please derive your method from object (see python old style classes)")
 
         if self.method is not None and type(new_method) is not type(self.method):
-            warnings.warn('Overriding database equation of state. From ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__, stacklevel=2)
+
+            # Warn user that they are changing the EoS
+            warnings.warn('Overriding database equation of state from ' + self.method.__class__.__name__ + ' to ' + new_method.__class__.__name__ + '. This does not overwrite any mineral attributes other than temperature and pressure. Non-implemented attributes of the old method will remain unchanged unless they are explicitly changed by the user.', stacklevel=2)
```

Comment:
If we change method from HP_TMT to MT, then it no longer makes any sense to have gibbs, Cp etc etc etc, but all of these attributes will be left unchanged from the previous set_state. This is a warning to the user so that they won't try to use any of thosee values after their next set_state. Eventually, we need to do as Ian suggests, but this warning is enough (IMO) for the current pull request.